index.d.ts

```diff
@@ -1,184 +1,261 @@
-import { Dispatch, SetStateAction, ReactNode } from 'react';
-
 declare module 'react-table' {
-    export interface Cell<D> {
-        render: (type: string) => any;
-        getCellProps: () => any;
-        column: Column<D>;
-        row: Row<D>;
-        state: any;
-        value: any;
-    }
+  import { ReactNode, ReactText, Dispatch, SetStateAction } from 'react'
+    export type SortingRule<D> = {
+    id: keyof D
+    desc: boolean
+  }
 
-    export interface Row<D> {
-        index: number;
-        cells: Cell<D>[];
-        getRowProps: () => any;
-        original: any;
-    }
+  export type SortingRules<D> = SortingRule<D>[]
 
-    export interface HeaderColumn<D, A extends keyof D = never> {
-        /**
-         * This string/function is used to build the data model for your column.
-         */
-        accessor: A | ((originalRow: D) => string);
```

Comment:
Agreed, we should probably use the type of `unknown` for the return value of the function. 

We should be able to properly represent the required `id` if the accessor is a function as well. Maybe worth doing after this PR just so we can get some types in that work.

Also, we should probably update the accessor function definition since its implementation is actually closer to this,

`(originalRow: D, index: number, { subRows: D[], depth: number, data: D[] }) => unknown`